### PR TITLE
Forward registers_on keyword arguments to gossip.register (closes #771)

### LINKED ISSUE #771: Make regsiters_on forward all unknown parameters to gossip.register

### PR DESCRIPTION
**Pull request by vmalloc**

### CODE REVIEW

File: slash/plugins/__init__.py [hunk 1]
```diff
@@ -1,14 +1,19 @@
+from collections import namedtuple
```
Comment (ayalash) [inline slash/plugins/__init__.py]: @vmalloc You added import to `namedtuple`, but you didn't used it

File: tests/test_plugins.py [hunk 2]
```diff
@@ -28,6 +28,32 @@ def some_method_here(self):
...
+            pass
+
+
+    @gossip.register('some.hook', provides=['some_requirement', 'other_requirement'])
+    def _unused():
+        pass
+
+    gossip.trigger('some.hook')
+    hook = gossip.get_hook('some.hook')
+    [r] = [r for r in hook.get_registrations() if r.func.__name__ == 'plugin_method']
```
Comment (ayalash) [inline tests/test_plugins.py]: Can we please change `r` to `reg` or something else? (`r` is return in pdb)

Comment (ayalash) [reply tests/test_plugins.py]: Do we want to check the `registers_on` plugin with `plugins.provides` and `plugins.needs` in class level (of the plugin) works as well?

**Comment by vmalloc** [comment]
@ayalash fixed and pushed. Please review again

**Pull request merged**